File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l=0
        r=len(matrix)-1
        while (l<=r):
            middle=(l+r)//2
            if (matrix[middle][0]<=target and matrix[middle][len(matrix[middle])-1]>=target):
                l=0
                r=len(matrix[middle])-1
                while l<=r:
                    lo=(l+r)//2
                    if matrix[middle][lo]==target:
                        return True
                    if matrix[middle][lo]>target:
                        r=lo-1
                    elif matrix[middle][lo]<target:
                        l=lo+1
                return False
            elif(target>matrix[middle][len(matrix[middle])-1]):
                l=middle+1
            else:
                r=middle-1
        return False
        

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> po = new HashMap<Integer,Integer>();
        for (int i=0; i <nums.length; i++){
            int cur = nums[i];
            int dif = target-cur;

            if (po.containsKey(dif))
            return new int[] {po.get(dif),i};


            po.put(nums[i],i);
        }   
       return new int[0];

        }     
    }

